src/java/org/apache/cassandra/metrics/TableMetrics.java

```diff
@@ -1173,32 +1204,55 @@ private LatencyMetrics createLatencyMetrics(String namePrefix, LatencyMetrics ..
      * @return true if first time metric with that name has been registered
      */
     private boolean register(String name, String alias, Metric metric)
+    {
+        return register(name, alias, null, metric);
+    }
+
+    /**
+     * Registers a metric to be removed when unloading CF.
+     * @param deprecated the deprecated name for the metric which should be {@code null} if no deprecation is required
+     * @return true if first time metric with that name has been registered
+     */
+    private boolean register(String name, String alias, String deprecated, Metric metric)
     {
         boolean ret = allTableMetrics.putIfAbsent(name, ConcurrentHashMap.newKeySet()) == null;
         allTableMetrics.get(name).add(metric);
-        all.add(() -> releaseMetric(name, alias));
+        all.add(() -> releaseMetric(name, alias, deprecated));
         return ret;
     }
 
-    private void releaseMetric(String metricName, String metricAlias)
+    private void releaseMetric(String metricName, String metricAlias, String metricDeprecated)
```

Comment:
I'd like to rename these from `String metricName, String metricAlias, String metricDeprecated` to something like `String tableMetricName, String cfMetricName, String tableMetricAlias`. It would at least try to match the reality of where the three of them live.